### PR TITLE
(PA-1272) Refactor our use of homebrew to not run as root

### PR DESCRIPTION
**Pull request by PopescuGheorghe**
WIP: changed project osx-10.12 template to use hombrew in the context of a test user
This is still work in progress but it would be good if you have a look on it.
Related PR's:
 https://github.com/puppetlabs/vanagon/pull/587
 https://github.com/puppetlabs/puppet-runtime/pull/113

### CODE REVIEW

**Comment by PopescuGheorghe** [comment]
Should the changes form [osx-10.12-x86_64.rb](https://github.com/puppetlabs/puppet-agent/pull/1568/files#diff-a0e6a4929f5fb452d2b75e02d4f50e38) to be also done on 6.0.x and master branch?

**Comment by caseywilliams** [comment]
Once https://github.com/puppetlabs/puppet-runtime/pull/113 is merged, it's true that the tarballs built with those changes will immediately affect all versions of puppet-agent, but unless the paths to the installed packages change somehow, I suspect that 6.0.x and master may still build successfully, since updates in puppet-runtime only affect the platform file. Have you tried a quick build of a 6.0.x or master agent with a runtime tarball built with https://github.com/puppetlabs/puppet-runtime/pull/113? If they can build, I'd say it's fine to wait for these changes to merge up naturally during nightly CI runs.

**Comment by PopescuGheorghe** [comment]
Tests were passing with the following configuration:
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/361/

File: configs/components/puppet-runtime.json [hunk 1]
```diff
@@ -1 +1 @@
-{"location":"http://builds.delivery.puppetlabs.net/puppet-runtime/201810110/artifacts/","version":"201810110"}
+{"location":"http://builds.delivery.puppetlabs.net/puppet-runtime/ff3a05c054651c1d1217deb8ebe3a82c9ae79f80/artifacts/","version":"201810110.1.gff3a05c"}
```
Comment (ScottGarman) [inline configs/components/puppet-runtime.json]: You'll need to exclude changes to puppet-runtime.json from this PR, as this file is auto-incremented by our CI.

**Comment by ScottGarman** [comment]
:+1:  provisionally with Casey's caveats.

**Comment by PopescuGheorghe** [comment]
@ScottGarman I reverted puppet-runtime.json

**Pull request merged**